lib/measures/default_feature_reports/measure.rb

```diff
@@ -203,6 +242,682 @@ def run(runner, user_arguments)
+    
+    ### net_site_energy
+    net_site_energy = sql_query(runner, sql_file, "AnnualBuildingUtilityPerformanceSummary", "TableName='Site and Source Energy' AND RowName='Net Site Energy' AND ColumnName='Total Energy'")
+    feature_report.reporting_periods[0].net_site_energy = convert_units(net_site_energy, 'GJ', 'kBtu')
+
+    ### net_source_energy
+    net_source_energy = sql_query(runner, sql_file, "AnnualBuildingUtilityPerformanceSummary", "TableName='Site and Source Energy' AND RowName='Net Source Energy' AND ColumnName='Total Energy'")
+    feature_report.reporting_periods[0].net_source_energy = convert_units(net_source_energy, 'GJ', 'kBtu')
+
+    ### net_utility_cost
+    #should be requested before simulation
+    
+    ### electricity
+    electricity = sql_query(runner, sql_file, "AnnualBuildingUtilityPerformanceSummary", "TableName='End Uses' AND RowName='Total End Uses' AND ColumnName='Electricity'")
+    feature_report.reporting_periods[0].electricity = convert_units(electricity, 'GJ', 'kBtu')
+
+    ### natural_gas
+    natural_gas = sql_query(runner, sql_file, "AnnualBuildingUtilityPerformanceSummary", "TableName='End Uses' AND RowName='Total End Uses' AND ColumnName='Natural Gas'")
+    feature_report.reporting_periods[0].natural_gas = convert_units(natural_gas, 'GJ', 'kBtu')
+
+    ### additional_fuel
+    additional_fuel = sql_query(runner, sql_file, "AnnualBuildingUtilityPerformanceSummary", "TableName='End Uses' AND RowName='Total End Uses' AND ColumnName='Additional Fuel'")
+    feature_report.reporting_periods[0].additional_fuel = convert_units(additional_fuel, 'GJ', 'kBtu')
+    
+    ### district_cooling
+    district_cooling = sql_query(runner, sql_file, "AnnualBuildingUtilityPerformanceSummary", "TableName='End Uses' AND RowName='Total End Uses' AND ColumnName='District Cooling'")
+    feature_report.reporting_periods[0].district_cooling = convert_units(district_cooling, 'GJ', 'kBtu')
+
+    ### district_heating
+    district_heating = sql_query(runner, sql_file, "AnnualBuildingUtilityPerformanceSummary", "TableName='End Uses' AND RowName='Total End Uses' AND ColumnName='District Heating'")
+    feature_report.reporting_periods[0].district_heating = convert_units(district_heating, 'GJ', 'kBtu')
+
+    ### water
+    water = sql_query(runner, sql_file, "AnnualBuildingUtilityPerformanceSummary", "TableName='End Uses' AND RowName='Total End Uses' AND ColumnName='Water'")
+    #feature_report.reporting_periods[0].water = convert_units(water, 'm3', 'ft3')
+    feature_report.reporting_periods[0].water = water
+
+    ### electricity_produced
+    electricity_produced = sql_query(runner, sql_file, "AnnualBuildingUtilityPerformanceSummary", "TableName='Electric Loads Satisfied' AND RowName='Total On-Site and Utility Electric Sources' AND ColumnName='Electricity'")
+    feature_report.reporting_periods[0].electricity_produced = convert_units(electricity_produced, 'GJ', 'kBtu')
+
+    ### end_uses
+    ## electricity
```

Comment:
You should review these measures:

https://github.com/NREL/OpenStudio-measures/blob/4bcbaa183d327cd1fd765c4b45e4c22a1381c537/nrel_published/XcelEDAReportingandQAQC/resources/EndUseBreakdown.rb

https://github.com/NREL/OpenStudio-measures/blob/a670520835fc1ffecc6e79c04dfc1fed3ba7519b/austin_published/enduse_timeseries_csv_export/measure.rb

https://github.com/NREL/OpenStudio-measures/blob/24b774ad8984e50f41f6baf7c967e1471ed82786/austin_published/austin_energy_eda_reporting_and_qaqc/resources/CreateResults.rb

Reply:
Might be some better ways to do this